src/coreclr/vm/jitinterface.cpp

```diff
@@ -3969,54 +3969,10 @@ CorInfoType CEEInfo::getTypeForPrimitiveValueClass(
     }
     else
     {
-        switch (th.GetInternalCorElementType())
+        CorElementType elementType = th.GetInternalCorElementType();
+        if (elementType >= ELEMENT_TYPE_BOOLEAN && elementType <= ELEMENT_TYPE_R8)
```

Comment:
This does not handle `ELEMENT_TYPE_I` and `ELEMENT_TYPE_U`

Reply:
Missed those, thanks.